### ISSUE TITLE
Syntax errors for valid GATs sytnax

### ISSUE THREAD

**Issue by LouayKamel** [rust-analyzer/rust-analyzer#9818]
`rust-analyzer` is rendering syntax errors.
```rust
impl SomeStruct {
    fn process<T>(v: T) -> <Self as GAT>::R<T>
    // Issue: rust-analyzer is complaining about syntax errors for valid GATs syntax.
    where Self: GAT<R<T> = T>
    {
        SomeStruct::do_something(v)
    }
}
```

Please have a look at https://github.com/rust-lang/rustfmt/issues/4943 for further details.

Note: please update your `rustfmt` from master branch to reproduce the bug.

**Comment by Veykril** [comment]
Actually this has been fixed already, make sure you are on the latest rust-analyzer, https://github.com/rust-analyzer/rust-analyzer/pull/9619

**Comment by LouayKamel** [comment]
@Veykril You're right, not sure why my atom ide-rust installed an old version in my `.local/bin` folder, however I replaced it with `new rust-analyzer binary` and it worked.
